addons/stock/models/stock_move.py

```diff
@@ -570,7 +627,7 @@ def _get_new_picking_values(self):
             'location_dest_id': self.location_dest_id.id,
         }
 
-    def _action_confirm(self):
+    def _action_confirm(self, merge=True):
         """ Confirms stock move or put it in waiting if it's linked to another move. """
```

Comment:
"""
:param: merge: According to this boolean, a newly confirmed move will be merged in another move of the same picking sharing its characteristics.
"""